projects/map-ol/src/lib/map-ol.service.ts

```diff
@@ -812,11 +892,106 @@ export class MapOlService {
     return newlayer;
   }
 
-  private create_wms_layer(l: WmsLayer): olTileLayer<olTileSource> {
+  private create_tms_layer(l: RasterLayer | VectorLayer): olTileLayer<olTileSource> {
+    let newlayer = null;
+    let newOlSource = null;
+    if (l instanceof RasterLayer) {
+      newlayer = this.create_xyz_layer(l);
```

Comment:
I used the xyz layer here because OpenLayers does not have no spezific TMS source.

https://openlayers.org/en/latest/apidoc/module-ol_source_XYZ-XYZ.html
For TMS the pased URL have to be in the correct format {-y}
I don't implemented custom tile grids with `tileUrlFunction`